assets/src/stories-editor/blocks/amp-story-page/edit.js

```diff
@@ -141,6 +150,33 @@ class EditPage extends Component {
 		return backgroundColorSettings;
 	}
 
+	ensureCTABeingLast() {
+		const {
+			getBlockOrder,
+			getBlock,
+			moveBlockToPosition,
+			clientId,
+		} = this.props;
+		const order = getBlockOrder( clientId );
+		if ( 1 >= order.length ) {
+			return;
+		}
+		let ctaClientID = false;
+		order.forEach( ( childId ) => {
+			const block = getBlock( childId );
+			if ( 'amp/amp-story-cta' === block.name ) {
```

Comment:
Speaking of the `innerBlocks.some....` check and `pageHasCTABlock`,  let's use that logic for this part here too. A function that could work for all these areas could be as simple as:

```js
const getCallToActionBlock( pageClientId ) => {
  const innerBlocks = getBlocksByClientId( getBlockOrder( pageClientId ) );
  return innerBlocks.find( ( { name } ) => name === 'amp/amp-story-cta' );
}
```

We could even use that function in `withCallToActionValidation`.